File: aulas-etecvav/3bim/Programação-e-algoritimo/python-conversion/cpp/ex09_fatorial.cpp
/*
    Autor: Kael Elodie Whitmore
    Data: Junho/2026
    Descrição: Calcula o fatorial de um número inteiro positivo.
*/

#include <iostream>
using namespace std;

int main()
{
    int numero;
    int contador = 1;
    int fatorial = 1;

    cout << "Digite um numero inteiro positivo: ";
    cin >> numero;

    if (numero >= 0)
    {
        while (contador <= numero)
        {
            fatorial = fatorial * contador;
            contador++;
        }

        cout << "Fatorial: " << fatorial << endl;
    }
    else
    {
        cout << "Numero invalido." << endl;
    }

    return 0;
}

File: aulas-etecvav/3bim/Programação-e-algoritimo/python-conversion/python/ex09_fatorial.py
# Autor: Matheus F. Mation
# Data: Agosto/2026 
# Descrição: Calcula o fatorial de um número inteiro positivo.

contador = 1
fatorial = 1

numero = int(input("Digite um numero inteiro positivo: "))

if numero >= 0:

    while contador <= numero:
        fatorial = fatorial * contador
        contador += 1 
    
    print(f"Fatorial: {fatorial}")

else:
    print("Numero invalido.")
